ire-adapter/src/main/scala/ingraph/ire/adapters/OneTimeQueryAdapter.scala

```diff
@@ -13,19 +14,20 @@ class IngraphOneTimeAdapter (
   val plan = FPlanParser.parse(querySpecification)
   val engine = EngineFactory.createQueryEngine(plan, indexer)
 
-  val transactionFactory = new TransactionFactory(16)
-  transactionFactory.subscribe(engine.inputLookup)
-  val transaction = transactionFactory.newBatchTransaction()
+  val dataSourceFactory = new DataSourceFactory
+  dataSourceFactory.subscribe(engine.inputLookup)
+  val dataSource = dataSourceFactory.newDataSource
   new PullTupleCreator(
     engine.vertexConverters.values.flatten.toVector.distinct,
     engine.edgeConverters.values.flatten.toVector.distinct,
-    indexer, transaction, LongIdParser)
+    indexer, dataSource, LongIdParser
+  )
 
   /**
     * Terminate execution and return results
     */
   def terminate(): Iterable[Tuple] = {
-    transaction.close()
+    dataSource.close()
```

Comment:
What happens when you close a datasource?

Reply:
Change sets are sent to subscribers: https://github.com/FTSRG/ingraph/blob/revamp-transactions/ire/src/main/scala/hu/bme/mit/ire/datasources.scala#L31-L36

Reply:
I know, but it makes less sense to close a datasource than to close a transaction.

Reply:
I renamed the type from 'transaction' because I found it misleading - 'transaction' suggests that you can run queries against it, not only add/remove tuples.
The `DataSource` trait is currently `AutoCloseable`, that's why this is called `close()`, but this could be changed easily. Any suggestions? (e.g. `DataSource#push()`?)

Reply:
I think DataSource is an ever weirder name for what it's doing. You add stuff to it, then you forward that data to Ire. `InputMultiplexer#sendAll`?

Reply:
But i honestly think Transaction is an ever better name for it.

>A transaction symbolizes a unit of work performed within a database management system (or similar system) against a database, and treated in a coherent and reliable way independent of other transactions.

This describe what we do perfectly. It's a set of additions and removals.

Reply:
The precise name would be `WriteOnlyTransaction` - even when it deletes, it doesn't read the data.
I like `InputMultiplexer` more.

Reply:
Yeah, we're diverging a bit from the colloquial usage and API of transactions, where you can run queries on them, but not from the formal definitions...

Reply:
I think the current state of the codebase shows why `close` is not a good name. For reading the data, we definitely do not close the "transaction" / "data source" / "input multiplexer", at least not in the sense as `AutoCloseable` puts it.